models/gif_psc_exp.h

```diff
@@ -0,0 +1,423 @@
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    double I_stim_; //!< This is piecewise constant external current
+    double V_;      //!< This is the membrane potential
+    double sfa_; //!< This is the change of the 'threshold' due to adaptation.
+    double stc_; //!< Spike triggered current.
+
+    std::vector< double > sfa_elems_; //!< Vector of adaptation parameters.
+    std::vector< double > stc_elems_; //!< Vector of spike triggered parameters.
+
+    double i_syn_ex_; //!< postsynaptic current for exc.
+    double i_syn_in_; //!< postsynaptic current for inh.
+
+    //!< absolute refractory counter (no membrane potential propagation)
+    unsigned int r_ref_;
+
+    State_(); //!< Default initialization
+
+    void get( DictionaryDatum&, const Parameters_& ) const;
+    void set( const DictionaryDatum&, const Parameters_& );
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Buffers of the model.
+   */
+  struct Buffers_
+  {
+    Buffers_( gif_psc_exp& );
+    Buffers_( const Buffers_&, gif_psc_exp& );
+
+    /** buffers and sums up incoming spikes/currents */
+    RingBuffer spikes_ex_;
+    RingBuffer spikes_in_;
+    RingBuffer currents_;
+
+    //! Logger for all analog data
+    UniversalDataLogger< gif_psc_exp > logger_;
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Internal variables of the model.
+   */
+  struct Variables_
+  {
+    double P30_;   // coefficient for solving membrane potential equation
+    double P33_;   // decay term of membrane potential
+    double P31_;   // coefficient for solving membrane potential equation
+    double P11ex_; // decay terms of excitatory synaptic currents
+    double P11in_; // decay terms of inhibitory synaptic currents
+    double P21ex_; // coefficient for solving membrane potential equation
+    double P21in_; // coefficient for solving membrane potential equation
+
+    std::vector< double >
+      P_sfa_; // decay terms of spike-triggered current elements
+    std::vector< double > P_stc_; // decay terms of adaptive threshold elements
+
+    librandom::RngPtr rng_; // random number generator of my own thread
+
+    unsigned int RefractoryCounts_;
+  };
+
+  // Access functions for UniversalDataLogger -----------------------
+
+  //! Read out the real membrane potential
+  double
+  get_V_m_() const
+  {
+    return S_.V_;
+  }
+
+  //! Read out the adaptive threshold potential
+  double
+  get_E_sfa_() const
+  {
+    return S_.sfa_;
+  }
+
+  //! Read out the spike triggered current
+  double
+  get_I_stc_() const
+  {
+    return S_.stc_;
+  }
+
+  double
+  get_input_currents_ex_() const
```

Comment:
If possible, could you replace ``get_input_currents_ex/in_`` by ``get_I_syn_ex/in_`` and rename ``S_.i_syn_ex_`` to ``S_.I_syn_ex_``? I'm sorry, I know this is a bit tedious, but since people tend to build on previous models, I'd rather have as many of them as coherent with the naming conventions as possible ;)